### PR TITLE
#163426629 Create Article Feature

### PR DESCRIPTION
**Pull request by eokwukwe**
#### What does this PR do?
User create article feature
#### Description of Task to be completed?
- [x] write an integration test for user create article feature
- [x] user can create an article
- [x] user can update and delete only their articles
- [x] user can view articles by other users
#### How should this be manually tested?
- [x] clone the repo
- [x] cd into the folder and run `npm install` to install dependencies
- [x] run `npm run test`
#### What are the relevant pivotal tracker stories?
[#163426629](https://www.pivotaltracker.com/story/show/163426629)

#### Screenshot
shot

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,11 +55,13 @@
     "passport-google-oauth20": "^1.0.0",
     "pg": "^7.8.0",
     "pg-hstore": "^2.3.2",
+    "reading-time-estimator": "^1.0.3",
     "request": "^2.87.0",
     "sequelize": "^4.42.0",
     "sinon": "^7.2.3",
     "slug": "^0.9.3",
     "underscore": "^1.9.1",
+    "uniqid": "^5.0.3",
```
Comment (benfluleck) [inline package.json]: This is from another commit please sort out your current branch

File: server/controllers/article.js [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+/**
+ * @description Get all published Articles
+ *
+ * @param {object} req HTTP request object
+ * @param {object} res HTTP response object
+ * @returns {object}  Response message object
+ */
+export const getAllArticles = async (req, res) => {
+  // const { limit, offset } = articlePagination(req.query);
```
Comment (benfluleck) [inline server/controllers/article.js]: Why is this comment here?

File: server/controllers/article.js [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+export const getArticleBySlug = async (req, res) => {
+  const { slug } = req.params;
+  try {
+    const article = await Articles.findOne({
+      where: {
+        slug: {
+          [Op.eq]: slug
+        }
+      },
+      // TODO
```
Comment (benfluleck) [inline server/controllers/article.js]: This is not  a todo comment `/* TODO: This is where the comment should be */`

**Review by Ndifreke — CHANGES_REQUESTED** [review]
Can you please squash your commit message history into a single commit

**Review by Ndifreke — APPROVED** [review]

**Review by Ndifreke — APPROVED** [review]

**Review by mrkingz — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: server/middlewares/validations/validations.js [hunk 4]
```diff
@@ -57,3 +60,41 @@ export const checkUniqueEmail = async (req, res, next) => {
...
+ * @description - Method to validate the query param for pagination
+ *
+ * @param {object} req - request sent to the server
+ * @param {object} res - response gotten from the server
+ * @param {function} next - callback function to continue execution
+ * @returns {object} - object representing response message
+ */
+export const validatePaginationParameters = (req, res, next) => {
+  const { page, limit } = req.query;
+  if (page && isNaN(page)) {
```
Comment (benfluleck) [inline server/middlewares/validations/validations.js]: Please rename this to pageNumber

File: server/middlewares/validations/validations.js [hunk 5]
```diff
@@ -57,3 +60,41 @@ export const checkUniqueEmail = async (req, res, next) => {
...
+    return res.status(400).json({
+      success: false,
+      message: "page can only be an Integer"
```
Comment (benfluleck) [inline server/middlewares/validations/validations.js]: Please rename to pageNumber

**Review by benfluleck — CHANGES_REQUESTED** [review]
Please rename page to pageNumber as it is more descriptive

File: server/middlewares/validations/validations.js (hunk 5 above)
Comment (eokwukwe) [reply server/middlewares/validations/validations.js]: Done

File: server/middlewares/validations/validations.js [hunk 6]
```diff
@@ -57,3 +60,41 @@ export const checkUniqueEmail = async (req, res, next) => {
...
+ * @param {object} res - response gotten from the server
+ * @param {function} next - callback function to continue execution
+ * @returns {object} - object representing response message
+ */
+export const validatePaginationParameters = (req, res, next) => {
+  const { pageNumber, limit } = req.query;
+  if (pageNumber && isNaN(pageNumber)) {
+    return res.status(400).json({
+      success: false,
+      message: "page can only be an Integer"
```
Comment (benfluleck) [inline server/middlewares/validations/validations.js]: In the message should be pageNumber as well

File: server/middlewares/validations/validations.js [hunk 7]
```diff
@@ -57,3 +60,41 @@ export const checkUniqueEmail = async (req, res, next) => {
...
+    });
+  }
+  if (limit && isNaN(limit)) {
```
Comment (benfluleck) [inline server/middlewares/validations/validations.js]: Change this to pageLimit please

File: server/middlewares/validations/validations.js [hunk 8]
```diff
@@ -57,3 +60,41 @@ export const checkUniqueEmail = async (req, res, next) => {
...
+    return res.status(400).json({
+      success: false,
+      message: "limit can only be an Integer"
```
Comment (benfluleck) [inline server/middlewares/validations/validations.js]: page limit please

**Review by benfluleck — CHANGES_REQUESTED** [review]

**Review by obasajujoshua31 — APPROVED** [review]

**Review by benfluleck — APPROVED** [review]

**Pull request merged**